tools/src/main/java/it/eng/tools/service/ApplicationPropertiesService.java

```diff
@@ -0,0 +1,243 @@
+				.map(c -> ApplicationProperty.Builder.newInstance()
+						.key(key)
+						.version((c.getVersion() != null ? c.getVersion() : 0))
+						.issued(c.getIssued())
+						.createdBy(c.getCreatedBy())
+						//.modified(Instant.now())
+						)
+				.orElseThrow(() -> new ApplicationPropertyNotFoundAPIException("ApplicationProperty with key: " + key + " not found"));
+	}
+	
+//	public Optional<ApplicationProperty> getStoredPropertyByKey(String key) {
+//		return repository.findById(key);
+//	}
+	
+	/*
+	 * public void deleteProperty(String name) { repository.deleteById(name); }
+	 */
+
+	/**
+	 * Private method for creating base builder for application property update by its ID.
+	 *
+	 * @param id The ID of the application property for update.
+	 * @return The builder for the application property  with basic mandatory unchanged fields.
+	 * @throws ApplicationPropertyErrorException Thrown if the application property  with the specified ID is not found.
+	 */
+	/*
+	 * public void updateProperty(String key, String value) {
+	 * 
+	 * System.out.println("\n\n\n" + env);
+	 * 
+	 * ConfigurableEnvironment configurableEnvironment = (ConfigurableEnvironment)
+	 * env; MutablePropertySources propertySources =
+	 * configurableEnvironment.getPropertySources();
+	 * 
+	 * PropertySource<?> ap = propertySources.get("applicationProperties"); if(ap !=
+	 * null) { System.out.println(ap.getSource().getClass().getName());
+	 * 
+	 * Map aaa = (Map)ap.getSource();
+	 * 
+	 * System.out.println(aaa.entrySet()); }
+	 * 
+	 * Map map = new HashMap<String,String>(); map.put(key, value);
+	 * 
+	 * propertySources.addFirst(new MapPropertySource("applicationProperties",
+	 * map));
+	 * 
+	 * MutablePropertySources ps = ((AbstractEnvironment) env).getPropertySources();
+	 * Iterator<PropertySource<?>> ips = ps.iterator();
+	 * 
+	 * while(ips.hasNext()) { PropertySource<?> currentps = ips.next();
+	 * 
+	 * String name = currentps.getName(); Object source = currentps.getSource();
+	 * System.out.println("\n\n\nname=" + name +
+	 * (currentps.getProperty("spring.ssl.bundle.jks.connector.keystore.location")
+	 * != null ?" YES":" NO")); System.out.println("\n" +
+	 * currentps.getClass().getName() + "\t" + currentps.toString());
+	 * System.out.println("\n" + source.getClass().getName() + "\t" +
+	 * source.toString() + "\n\n\n"); }
+	 * 
+	 * }
+	 */
+
+	/*
+	 * public void saveAllPropertiesOnEnv() { ConfigurableEnvironment
+	 * configurableEnvironment = (ConfigurableEnvironment) env;
+	 * MutablePropertySources propertySources =
+	 * configurableEnvironment.getPropertySources(); PropertySource<?>
+	 * customPropertySource = propertySources.get("customPropertySource"); if
+	 * (customPropertySource != null) { // Write properties to configuration file //
+	 * For example, write to application.properties
+	 * 
+	 * //TODO: manage add on env when customPropertySource exists } }
+	 */
+
+	public String addPropertyToApplicationPropertySource(String key, Object value) {

```

Comment:
Where is this method used?
Search does not find it or maybe I did not look good